### PR TITLE
Update all Jest packages to 17.x

### PR DESCRIPTION
**Pull request by yaycmyk**
~~The caching is mostly for CI speediness, users won't see more than a few seconds of improvement.~~ It did not help in any useful way, so dropped that portion of the PR.

### CODE REVIEW

**Comment by gaearon** [comment]
I can just remove the CI caches, would that help?

**Comment by yaycmyk** [comment]
It's all good @gaearon, from what I can see the impact is negligible and it causes first run to be 50% slower which is poor DX.

**Comment by yaycmyk** [comment]
Looks like it's getting stuck on the fiber tests. I'm without a computer for a few days, but will take a look on Saturday.

File: package.json [hunk 1]
```diff
@@ -58,10 +59,11 @@
     "gulp-load-plugins": "^1.2.4",
     "gulp-util": "^3.0.7",
     "gzip-js": "~0.3.2",
-    "jest": "^15.1.1",
-    "jest-config": "^15.1.1",
-    "jest-jasmine2": "^15.1.1",
-    "jest-runtime": "^15.1.1",
+    "jest": "^17.0.2",
+    "jest-cli": "17.0.2",
```
Comment (keyanzhang) [inline package.json]: `jest` just exports `jest-cli` (https://github.com/facebook/jest/blob/master/packages/jest/src/jest.js) so only one of them would be enough here.

**Comment by joh1234** [comment]
Update README
https://goo.gl/JBljdR

or 
http://bit.ly/2dK6Ojk

File: scripts/fiber/record-tests [hunk 2]
```diff
@@ -8,8 +8,9 @@ const fs = require('fs');
 const os = require('os');
 const path = require('path');
 
-const SearchSource = require('jest').SearchSource;
-const TestRunner = require('jest').TestRunner;
+const SearchSource = require('jest-cli').SearchSource;
+const TestRunner = require('jest-cli').TestRunner;
+const TestWatcher = require('jest-cli/build/TestWatcher');
```
Comment (yaycmyk) [inline scripts/fiber/record-tests]: There is a PR open to remove the need to delve into internals. https://github.com/facebook/jest/pull/2138

**Comment by yaycmyk** [comment]
@gaearon / @keyanzhang this should be good now. Thanks for the feedback!

File: package.json [hunk 3]
```diff
@@ -61,10 +62,10 @@
     "gulp-load-plugins": "^1.2.4",
     "gulp-util": "^3.0.7",
     "gzip-js": "~0.3.2",
-    "jest": "^15.1.1",
-    "jest-config": "^15.1.1",
-    "jest-jasmine2": "^15.1.1",
-    "jest-runtime": "^15.1.1",
+    "jest-cli": "17.0.2",
```
Comment (keyanzhang) [inline package.json]: Sorry, can you change it back to `jest`? It's being used here: https://github.com/facebook/react/blob/64cba04bf976f2f11b428bb9ec1108cd2da70e53/scripts/fiber/record-tests#L11

**Comment by yaycmyk** [comment]
I already fixed it. See the 2 most recent PRs.

**Comment by gaearon** [comment]
Looks great, thanks!

**Pull request merged**